### ISSUE TITLE
[feature] Do you mind a CI/CD to update AUR automatically?

### ISSUE THREAD

**Issue by Freed-Wu** [priv-kweihmann/oelint-parser#208]
<https://aur.archlinux.org/packages/python-oelint-parser>

**Comment by priv-kweihmann** [comment]
As the account over on AUR seems to be yours, I have a different proposal to get it done,
also because I avoided creating distro packages in the past, due to the missing testing capabilities.

How about on every release I send out a trigger to a repo pipeline of yours and could do the packaging.
The only thing I would need is a either being a collaborator in the repo of yours (then I could use my own tokens) or you somehow send me a super limited token of yours (preferably by mail or something).

This trigger would be charming in a way, that it could be extended to other packaging pipelines (rpm, deb immediately come to my mind), without giving me the need to test them here as part of this repo + the benefit that the access credentials to AUR can stay with you.

See https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#repository_dispatch.

Let me know what you think

**Comment by Freed-Wu** [comment]
You can refer <https://github.com/lyokha/g3kb-switch/issues/24>. <https://github.com/lyokha/g3kb-switch/blob/master/.github/workflows/main.yml> provide deb, rpm and upload AUR automatically. Is this project can accept a CI/CD like it? And you need to register an AUR account and let me add you to the project of AUR.

**Comment by priv-kweihmann** [comment]
As said, I would prefer to "outsource" building any packages for distros, so my idea is

```mermaid
sequenceDiagram
    oelint-adv->>oelint-adv: release
    oelint-adv--xdownstream-packager: release signal
    downstream-packager-)+downstream-packager: build package
    downstream-packager-)-downstream-packager: test package
    downstream-packager->>X: upload to package feed (e.g. AUR)
```

So this project doesn't need to hold credentials for all kinds of package feeds, is extendable, while still having everything as automated as possible.

In my idea the `release signal` will be just a json holding the new version, just being published to pypi and from there downstream packager can pick it up and do there arch/distro specific packaging

**Comment by Freed-Wu** [comment]
> How about on every release I send out a trigger to a repo pipeline of yours and could do the packaging.

Oh, I know your meaning.

To the best of my knowledge, the downstream have a CI/CD, then detect if the upstream is updated, then fetch new version. In fact, many project use this model. The upstream don't need any change, such as a CI/CD to send signal.

**Comment by priv-kweihmann** [comment]
So in case you agree with the proposed model and I should implement a trigger, just let me know where to send the trigger to (repo URL or something) - then I gladly do it

**Comment by Freed-Wu** [comment]
What is trigger? Is it adding a job to github workflow like:

```diff
diff --git a/.github/workflows/autorelease.yml b/.github/workflows/autorelease.yml
index 3844f51..d37cdc3 100644
--- a/.github/workflows/autorelease.yml
+++ b/.github/workflows/autorelease.yml
@@ -119,3 +119,9 @@ jobs:
       with:
         user: __token__
         password: ${{ secrets.PYPI_API_TOKEN }}
+
+    - name: Trigger AUR update
+      uses: Freed-Wu/update-aur-package@v1.0.11
+      with:
+        package_name: python-oelint-parser
+        ssh_private_key: ${{secrets.AUR_SSH_PRIVATE_KEY}}
```

?

**Comment by priv-kweihmann** [comment]
not quite...

Let's say you have a repository `myusername/myrepo` on github.
Then in this repo you have a file under `.github/workflows` and let's name it `pipeline.yml`.
In this file you would have something like this

```
name: My pipeline

on: 
  repository_dispatch:
     types: newversion
jobs:
  test:
    name: Test
    runs-on: ubuntu-latest
    steps:
      - run: 'echo "field: ${{ github.event.client_payload.version }}"'
```

and now I can call this pipeline with

```
curl -X POST https://api.github.com/repos/myusername/myrepo/dispatches \
          -H 'Accept: application/vnd.github.everest-preview+json' \
          -u ${{ secrets.ACCESS_TOKEN }} \
          -d '{"event_type": "newversion", "client_payload": {"version": "1.2.3"}}'
```

here from this repository and you would have the information there is a new version `1.2.3` available in the downstream repo and can do all the packaging with the autorelease workflow as you described already.
The thing I would need is the information `myusername/myrepo` and `newversion` (from types attribute) and I could send you a trigger every time a new version is released here.

Let me see if I can find the time on the weekend to come up with a draft

**Comment by Freed-Wu** [comment]
This looks like the mode of NUR, not AUR.

<https://github.com/nix-community/nur?tab=readme-ov-file#update-nurs-lock-file-after-updating-your-repository>

Currently I haven't have repo for AUR in github :cry:. I think we should wait more packagers of other Distributions in order to see their arguments.
